backend/core/common/get_current_user.go

```diff
@@ -0,0 +1,28 @@
+package common
+
+// there's probably a better place for this file but i suck at system design so whatever
+// deal with it :)
+// nah but fr i need help where do i put this??? 😭😭😭😭
+
+import (
+	"backend/core/db/models"
+
+	"github.com/gin-gonic/gin"
+	"gorm.io/gorm"
+)
+
+func GetCurrentUser(ctx *gin.Context, userId string) *models.User {
+	db, _ := ctx.MustGet("db").(*gorm.DB)
+
+	var foundUser models.User
+	userQueryResult := db.
+		// Preload the congregation to ensure it's included in the user payload
+		Preload("Congregation").
+		First(&foundUser, "id = ?", userId)
+
+	if userQueryResult.Error != nil {
+		return nil
+	}
+
+	return &foundUser
```

Comment:
To the rescue! We're using something called "service layer architecture". Because we are separating business logic (i.e. services) from presentation layer (i.e. views and API endpoints).

In this pattern a common practice is to store data access code into something called repositories. So this counts as a repository function since it's doing data access from the database. Now that we are separating database access, we should put them into a folder called 'repository' in the 'db' folder.

Reply:
This makes a lot of sense now... need to read up on all this architecture stuff